test/test_rhat.py

```diff
@@ -33,11 +41,117 @@ def test_rhat_ragged():
     np.testing.assert_allclose(rhat_expected(chains), rhat(chains), atol=0.1, rtol=0.2)
 
 
-def test_rhat_exceptions():
-    with pt.raises(ValueError):
-        chains = []
-        rhat(chains)
-    with pt.raises(ValueError):
-        chain1 = [1.01, 1.2, 1.3, 1.4]
-        chains = [chain1]
-        rhat(chains)
+def test_rhat_size_exceptions():
+    bad1 = []
+    bad2 = ([[1.01, 1.2, 1.3, 1.4]],)
+    bad3 = [[1, 2, 3], [4], [5, 6, 7, 8, 9]]
+    for chains in [bad1, bad2, bad3]:
+        with pt.raises(ValueError):
+            rhat(chains)
+
+
+def test_split_chains():
+    np.testing.assert_equal([], split_chains([]))
+    np.testing.assert_equal([[1], []], split_chains([[1]]))
+    np.testing.assert_equal([[1], [2]], split_chains([[1, 2]]))
+    np.testing.assert_equal([[1, 2], [3]], split_chains([[1, 2, 3]]))
+    np.testing.assert_equal([[1, 2, 3], [4, 5, 6]], split_chains([[1, 2, 3, 4, 5, 6]]))
+    np.testing.assert_equal(
+        [[1, 2], [3], [4, 5], [6, 7]], split_chains([[1, 2, 3], [4, 5, 6, 7]])
```

Comment:
Why's it decide 3 has to sit by itself? (The example in the docstring said it'd be 7 that's isolated.)

Is there some variation in implementations or something?

Reply:
Ah, okay! But what about the `[[1, 2, 3], [4, 5, 6, 7]]` example? Ultimately this is an implementation detail that doesn't really matter, and probably doesn't need to be documented; I'm just curious.

Reply:
I added `[[1, 2, 3], [4, 5, 6, 7]]` as a test case and added it as an example in the doc.

This implementation detail matters in the sense that the answer will be different with two different splits.